ykrt/src/mt.rs

```diff
@@ -410,9 +410,42 @@ impl MTThread {
 }
 
 /// What action should a caller of `MT::transition_location` take?
+#[derive(Debug, PartialEq)]
 enum TransitionLocation {
     NoAction,
     Execute(*const CompiledTrace),
     StartTracing(TracingKind),
     StopTracing(*const HotLocation),
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::location::HotLocationDiscriminants;
+
+    fn hotlocation_discriminant(loc: &Location) -> Option<HotLocationDiscriminants> {
+        match loc.lock() {
+            Ok(ls) => {
+                let x = HotLocationDiscriminants::from(&*unsafe { ls.hot_location() });
+                loc.unlock();
+                Some(x)
+            }
+            Err(()) => None,
```

Comment:
To follow suit with the `Ok` case, shouldn't the `Err` case be:

```
Err(()) => {
    loc.unlock();
    None
}
```

?

Although since this is such a tiny critical section, I'd be inclined to just put a single `unlock()` outside of the match.

Reply:
If `Err` was returned a loc wasn't obtained.

Reply:
s/loc/lock/

Reply:
ah yes, my bad.